src/Support/RulesInFields.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rebing\GraphQL\Support;
+
+use GraphQL\Error\InvariantViolation;
+use GraphQL\Type\Definition\ResolveInfo;
+use GraphQL\Type\Definition\Type;
+use GraphQL\Type\Definition\WrappingType;
+
+class RulesInFields
+{
+    /**
+     * @var  Type
+     */
+    private $parentType;
+    /**
+     * @var ResolveInfo
+     */
+    private $resolveInfo;
+
+    /**
+     * @var array<mixed,mixed>
+     */
+    private $fieldsAndArguments;
+
+    /**
+     * @param Type $parentType
+     * @param ResolveInfo $resolveInfo
+     */
+    public function __construct(Type $parentType, ResolveInfo $resolveInfo)
+    {
+        $this->parentType = $parentType;
+        $this->resolveInfo = $resolveInfo;
+        $this->fieldsAndArguments = (new ResolveInfoFieldsAndArguments($this->resolveInfo))->getFieldsAndArgumentsSelection(5);
+    }
+
+    /**
+     * @return array<array>
+     */
+    public function get(): array
+    {
+        if ($this->parentType instanceof WrappingType) {
+            $this->parentType = $this->parentType->getWrappedType(true);
+        }
+
+        return [$this->fieldsAndArguments, $this->getRules($this->fieldsAndArguments, null, $this->parentType)];
+    }
+
+    /**
+     * @param  array<string,mixed>|string|callable  $rules
+     * @param  array<string,mixed>  $arguments
+     * @return array<string,mixed>|string
+     */
+    protected function resolveRules($rules, array $arguments)
+    {
+        if (is_callable($rules)) {
+            return call_user_func($rules, $arguments, $this->fieldsAndArguments);
```

Comment:
What would be the use case to provide the full arguments too?

I see that closure and the test, but they don't make meaningful use of it